.github/workflows/test.yaml

```diff
@@ -95,3 +95,50 @@ jobs:
         run: |
           conda activate test-environment
           codecov
+  ui_test_suite:
+    name: UI tests on ${{ matrix.os }} with Python 3.9
+    needs: [pre_commit]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
```

Comment:
Our experience on Panel shows that it's easy to write flaky tests. The more UI tests you run, the more likely it is one of these tests will fail. Because of that, and because I don't think the OS is so relevant to UI tests (to me these are integration tests, that shouldn't cover so much code that isn't covered by unit tests), I'd really consider just running theses tests on one OS, and I'd pick ubuntu as it's the fastest!

(The browser tested is maybe more relevant than the OS for UI tests, let's not get into that now though :) )

Reply:
One OS is fine by me.